packages/purgecss/src/index.ts

```diff
@@ -214,17 +214,28 @@ function isTagFound(
 }
 
 /**
- * Returns true if the selector is inside a pseudo class
+ * Returns true if the node is a pseudo class
  * (e.g. :nth-child, :nth-of-type, :only-child, :not)
- * @param selector selector
+ * @param node node
  */
-function isInPseudoClass(selector: selectorParser.Node): boolean {
-  return (
-    (selector.parent &&
-      selector.parent.type === "pseudo" &&
-      selector.parent.value.startsWith(":")) ||
-    false
-  );
+function isPseudoSelector(node: selectorParser.Node): boolean {
+  return node.type === "pseudo" && node.value.startsWith(":");
+}
+
+/**
+ * Returns true if the node is inside a pseudo class
+ * (e.g. :nth-child, :nth-of-type, :only-child, :not)
+ * @param node selector
+ */
+function isInPseudoClass(node: selectorParser.Node): boolean {
```

Comment:
I think it is safe to change the node to `selectorParser.Selector` here

Reply:
I just tried to not break the existing interface.

It looks like it works like a charm.

Thanks for the notice!